python-backend/tests/document_manager/resources/test_document_manager_resource.py

```diff
@@ -1,26 +1,59 @@
-import json
-import io
-import filecmp
-import os
-import pytest
-import shutil
+import json, uuid, os
+from unittest import mock
 
+from app.extensions import cache
+from app.api.constants import DOWNLOAD_TOKEN
+from tests.factories import DocumentManagerFactory
 
-def test_download_file_no_guid(test_client, db_session, auth_headers):
-    get_resp = test_client.get(
-        f'/document-manager', headers=auth_headers['full_auth_header'])
+
+def test_download_file_happy_path(test_client, db_session, auth_headers, tmp_path):
+    document = DocumentManagerFactory(path_root=tmp_path, file_display_name='testfile.pdf')
+
+    test_data = 'Contents of file'
+    file_path = document.full_storage_path
+    os.makedirs(os.path.dirname(file_path), exist_ok=True)
+    with open(file_path, "w") as f:
+        f.write(test_data)
+
+    token_resp = test_client.get(
+        f'/document-manager/{document.document_guid}/token',
+        headers=auth_headers['full_auth_header'])
+    token_data = json.loads(token_resp.data.decode())
+    token_guid = token_data['token_guid']
+    assert token_resp.status_code == 200
+    assert token_guid
+
+    with mock.patch.object(cache, 'get') as mock_cache_get:
+        mock_cache_get.return_value = document.document_guid
+
+        get_resp = test_client.get(f'/document-manager?token={token_guid}')
+        assert get_resp.status_code == 200
+        assert get_resp.data.decode() == test_data
+        mock_cache_get.assert_called_with(DOWNLOAD_TOKEN(token_guid))
+
+
+def test_download_file_no_token(test_client, db_session):
+    get_resp = test_client.get(f'/document-manager')
+    get_data = json.loads(get_resp.data.decode())
+
+    assert get_resp.status_code == 400
+    assert get_data['error']['status'] == 400
+    assert get_data['error']['message'] is not ''
+
+
+def test_download_file_invalid_token(test_client, db_session):
+    get_resp = test_client.get(f'/document-manager?token={uuid.uuid4()}')
     get_data = json.loads(get_resp.data.decode())
 
     assert get_resp.status_code == 400
     assert get_data['error']['status'] == 400
     assert get_data['error']['message'] is not ''
 
 
-def test_download_file_no_doc_with_guid(test_client, db_session, auth_headers):
-    get_resp = test_client.get(
-        f'/document-manager/1234', headers=auth_headers['full_auth_header'])
+def test_get_file_with_guid(test_client, db_session):
+    get_resp = test_client.get(f'/document-manager/1234')
     get_data = json.loads(get_resp.data.decode())
 
-    assert get_resp.status_code == 404
-    assert get_data['error']['status'] == 404
+    assert get_resp.status_code == 500
+    assert get_data['error']['status'] == 500
```

Comment:
What's the reason for making this a 500?

Reply:
This is now a 500 because the test is hitting an API path that doesn't exist. The download path used to be 'document-manager/GUID' but now its 'document-manager?token=token'

Reply:
Oh, okay. Any thoughts on when it would make sense to remove tests for deprecated endpoints?

I took a quick look at the spec (https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html), and it looks like we might want code 410 instead? But I'm not super picky about that either way

Reply:
I've deleted the test because it has no real value.